src/frontend/Ast_to_Mir.ml

```diff
@@ -352,12 +352,12 @@ let read_decl dread decl_id transform sizedtype smeta decl_var =
     | Indexed (_, indices) -> {var with expr= Indexed (readfn var, indices)}
     | _ -> readfn var
   in
-  let st =
+  let forl, st =
     match dread with
-    | ReadData -> sizedtype
-    | ReadParam -> param_size transform sizedtype
+    | ReadData -> (for_scalar, sizedtype)
```

Comment:
This is really interesting! So the reading of data just iterates over eigen-types, rather than doing something efficient using Eigen library functions?

Reply:
Did you have an Eigen library function in mind?

Reply:
Not really. What I am mostly surprised by is to see code like
```
for (i in 1:100)
  mat[i] = some_possibly_expensive_function_call()[i]
```

Reply:
In this case we don't have that expensive call, but I think it's basically up to code gen to figure that out and do something like instruction selection here where it can see the entire for loop and do the right thing.